rules-tests/TypeDeclaration/Rector/Property/TypedPropertyFromAssignsRector/FixturePhp82/null_default_value.php.inc

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\TypedPropertyFromAssignsRector\FixturePhp82;
+
+final class NullDefaultValue
+{
+    private $isEnabled = null;
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Property\TypedPropertyFromAssignsRector\FixturePhp82;
+
+final class NullDefaultValue
+{
+    private null $isEnabled = null;
```

Comment:
I don't think it makes sense to have this type at property level. 

It fits better as return type or part of a union

Reply:
The similar type map exists on Boolean already, see:

https://github.com/rectorphp/rector-src/blob/72815588219a43f3881295c77bd5c97239e4721a/packages/PHPStanStaticTypeMapper/TypeMapper/BooleanTypeMapper.php#L52-L63

and applied for false/true at 

https://github.com/rectorphp/rector-src/blob/72815588219a43f3881295c77bd5c97239e4721a/rules-tests/TypeDeclaration/Rector/Property/TypedPropertyFromAssignsRector/FixturePhp82/false_default_value.php.inc#L5-L19

It improve the type coverage when no type at all rather than just mixed without type.

The usage of removal of property as always never changed is based on the user perspective of what design to apply.

Reply:
I agree with @staabm , as these types are expected to change to `false/true` or `null` + something. 

In these case fixture should skip the `null` and intsead of `true`/`false` use `bool`. We cannot know what is the user intention and it's rarely a single type, as that would be always the same and removed.

Reply:
Ok, I updated to only apply standalone null null,false,true on return type https://github.com/rectorphp/rector-src/pull/4783/commits/afbaa1785ee580452798636b5dd9ef9d310a940f 👍